Add networking

briankfitzwater opened the pull request:
- copied networking files from grace-core repository
- made some changes to correct list vs string errors
- changed the mgmt_account_id and env_account_id variables to match grace-core
- fix circle-env-deployer role to allow circle-mgmt-deployer user to assume role
- adds terraform variables for networking as an AES-256 encrypted file

Commit:
WIP: Copied files for grace-core

Commit:
WIP: Need to fix access role

Commit:
WIP: Change account id variable names

Commit:
WIP: Allow circle-mgmt-deployer user to assume circle-env-deployer role

Commit:
Encrypt/Decrypt networking terraform variables

briankfitzwater commented:
![](https://github.trello.services/images/mini-trello-icon.png) [DEV - Implement GRACE network architecture in grace-example](https://trello.com/c/ZWEfBSAs/633-dev-implement-grace-network-architecture-in-grace-example)

ManojChalise reviewed (commented):
I would really like to see those Security Group and NACL untangled from VPC module. We will need some governance of those perimeter stuff and we need to figure out, how to do it. For that reason i think we should keep things like SG, NACL, URL list etc separate from other configuration. 

Also i think this environment will be connected to GSA. But since it's not prod, i am worried about having from wide rules like from/to 0.0.0.0 and someone start picking up and using it, right now. May be we should limit those for now.

Commit:
Remove security groups and NACLs

Commit:
Add statement about networking to Terrafrom README

briankfitzwater commented:
@ManojChalise I removed the security group and NACL configurations.  I added sub-tasks to the [grace-example epic card](https://trello.com/c/Pqpt0DEC/575-dev-create-grace-example-to-demonstrate-the-grace-architecture-and-best-practices-epic) in Trello.
